agent/tests/test_php_wrapper.c

```diff
@@ -85,6 +90,460 @@ NR_PHP_WRAPPER(test_add_2_arrays) {
+      test_name_txn_before_not_ok, NULL, test_name_txn_before_not_ok, NULL,
+      test_name_txn_before_ok, NULL, "three",
+      "one:beforenotok,two:beforenotok,three:beforeok");
+
+  /*
+   * special function for one is called before and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `two`
+   */
+  setup_nested_framework_calls(
+      test_name_txn_before_not_ok, NULL, test_name_txn_before_ok, NULL,
+      test_name_txn_before_not_ok, NULL, "two",
+      "one:beforenotok,two:beforeok,three:beforenotok");
+
+  /*
+   * special function for one is called before and uses NR_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `three`
+   */
+  setup_nested_framework_calls(test_name_txn_before_ok, NULL,
+                               test_name_txn_before_not_ok, NULL,
+                               test_name_txn_before_ok, NULL, "three",
+                               "one:beforeok,two:beforenotok,three:beforeok");
+
+  /*
+   * special function for one is called before and uses NR_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `one`
+   */
+  setup_nested_framework_calls(
+      test_name_txn_before_ok, NULL, test_name_txn_before_not_ok, NULL,
+      test_name_txn_before_not_ok, NULL, "one",
+      "one:beforeok,two:beforenotok,three:beforenotok");
+
+  /*
+   * special function for one is called before and uses NR_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `two`
+   */
+  setup_nested_framework_calls(test_name_txn_before_ok, NULL,
+                               test_name_txn_before_ok, NULL,
+                               test_name_txn_before_not_ok, NULL, "two",
+                               "one:beforeok,two:beforeok,three:beforenotok");
+
+  /*
+   * Before/After and NR_NOT_OK_TO_OVERWRITE/NR_NOT_OK_TO_OVERWRITE mixes
+   */
+
+  /*
+   * special function for one is called after and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `three`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_not_ok,
+                               test_name_txn_before_ok, NULL,
+                               test_name_txn_before_ok, NULL, "three",
+                               "one:afternotok,two:beforeok,three:beforeok");
```

Comment:
So the call stack is:
```
one
  before_callback_for_two
    nr_txn_set_path sets txn name to 'two' because it is not set yet
  two
    before_callback_for_three
      nr_txn_set_path sets txn name to 'three' because called with NR_OK_TO_OVERWRITE
    three
after_callback_for_one
  nr_txn_set_path leaves txn name as 'three' because called with NR_NOT_OK_TO_OVERWRITE
```

Reply:
please see: [857fa16](https://github.com/newrelic/newrelic-php-agent/pull/615/commits/857fa1642e98a49e0a3b46e1404af1b6a4146637)
Convention is to describe what sort of wrapper each function is calling.